File: algorithm/src/main/java/baekjoon/p02954/Main.java
package baekjoon.p02954;

import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println(solve(sc.nextLine()));
    }

    private static String solve(String input) {
        StringBuilder sb = new StringBuilder();

        for (int i = 0; i < input.length(); i++) {
            char ch = input.charAt(i);
            sb.append(ch);

            if (vowel(ch)) {
                i += 2;
            }
        }

        return sb.toString();
    }

    private static boolean vowel(char ch) {
        switch (ch) {
            case 'a':
            case 'e':
            case 'i':
            case 'o':
            case 'u':
                return true;
            default:
                return false;
        }
    }

}

File: algorithm/src/main/java/baekjoon/p02954/Main.kt
package baekjoon.p02954

fun main(args: Array<String>) {
    println(solve(readLine()!!))
}

private fun solve(input: String): String {
    return buildString {
        var i = 0
        while (i < input.length) {
            val ch = input[i]
            append(ch)

            when (ch in "aeiou") {
                true -> i += 3
                false -> i++
            }
        }
    }
}